internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -839,13 +831,11 @@ func validateCreateRequest(ctx context.Context, req *pbs.CreateAuthMethodRequest
 		}
 		switch subtypes.SubtypeFromType("auth", req.GetItem().GetType()) {
 		case password.Subtype:
-			attrs := &pb.PasswordAuthMethodAttributes{}
-			if err := handlers.StructToProto(req.GetItem().GetAttributes(), attrs); err != nil {
-				badFields[attributesField] = "Attribute fields do not match the expected format."
-			}
+			// Password attributes are not required when creating a password auth method.
 		case oidc.Subtype:
-			attrs := &pb.OidcAuthMethodAttributes{}
-			if err := handlers.StructToProto(req.GetItem().GetAttributes(), attrs); err != nil {
+			attrs := req.GetItem().GetOidcAuthMethodsAttributes()
+			if attrs == nil {
+				// OIDC attributes are required when creating an OIDC auth method.
```

Comment:
I am wondering if the error message returned here should be updated. The specific error that was being checked here will now happen in the interceptor, but it is possible that no attributes are provided at all, which would reach this code. I suppose it is technically correct to say that `nil` is that the expected format, but it seems a little confusing to me.

Reply:
Done in https://github.com/hashicorp/boundary/pull/1972/commits/7899545d25d153bef7a0a7e80e3c03d54a45a815